idea/data-storage/src/gear/gear.service.ts

```diff
@@ -0,0 +1,325 @@
+      console.log(error);
+    }
+  }
+
+  private async handleExtrinsics(block: SignedBlockExtended, timestamp: number) {
+    const status = this.api.createType('ExtrinsicStatus', { finalized: block.block.header.hash.toHex() });
+
+    await this.handleCodeExtrinsics(block, status, timestamp);
+
+    await this.handleProgramExtrinsics(block, status, timestamp);
+
+    await this.handleMessageExtrinsics(block, status, timestamp);
+  }
+
+  private async handleMessageExtrinsics(block: SignedBlockExtended, status: ExtrinsicStatus, timestamp: number) {
+    const extrinsics = getExtrinsics(block, ['sendMessage', 'sendReply', 'uploadProgram', 'createProgram']);
+    if (extrinsics.length === 0) {
+      return;
+    }
+
+    const messages: Message[] = [];
+
+    for (const tx of extrinsics) {
+      const foundEvent = filterEvents(tx.hash, block, block.events, status).events.find(({ event }) =>
+        this.api.events.gear.MessageQueued.is(event),
+      );
+
+      if (!foundEvent) {
+        continue;
+      }
+
+      const {
+        data: { id, source, destination, entry },
+      } = foundEvent.event as MessageQueued;
+
+      const [payload, value] = getPayloadAndValue(tx.args, tx.method.method);
+
+      messages.push(
+        plainToInstance(Message, {
+          id: id.toHex(),
+          destination: destination.toHex(),
+          source: source.toHex(),
+          payload,
+          value,
+          timestamp: new Date(timestamp),
+          genesis: this.genesis,
+          program: await this.programRepository.get(destination.toHex(), this.genesis),
+          type: MessageType.ENQUEUED,
+          entry: entry.isInit
+            ? MessageEntryPoint.INIT
+            : entry.isHandle
+              ? MessageEntryPoint.HANDLE
+              : MessageEntryPoint.REPLY,
+        }),
+      );
+
+      await this.messageService.createMessages(messages);
+    }
+  }
+
+  private async handleProgramExtrinsics(block: SignedBlockExtended, status: ExtrinsicStatus, timestamp: number) {
+    const extrinsics = getExtrinsics(block, ['uploadProgram', 'createProgram']);
+
+    if (extrinsics.length === 0) {
+      return;
+    }
+
+    const programs: Program[] = [];
+
+    for (const tx of extrinsics) {
+      const foundEvent = filterEvents(tx.hash, block, block.events, status).events.find(({ event }) =>
+        this.api.events.gear.MessageQueued.is(event),
+      );
+
+      if (!foundEvent) {
+        continue;
+      }
+
+      const {
+        data: { source, destination },
+      } = foundEvent.event as MessageQueued;
+      const programId = destination.toHex();
+      const owner = source.toHex();
+      const blockHash = block.block.hash.toHex();
+
+      const codeId = tx.method.method === 'uploadProgram' ? generateCodeHash(tx.args[0].toHex()) : tx.args[0].toHex();
+      let code: Code;
+      try {
+        code = await this.codeRepository.get(codeId, this.genesis);
+      } catch (e) {
```

Comment:
Can you please also rename `e` to `error`?  in most of the places, we have already used the `error` name.

Reply:
There is nothing to rename. I've deleted this part